#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.3578011255384946 -0.0619622094947228 0.9317397915501509 3.461551011991932
  position 6.086154018252933 0.5547562830495163 5.5167809553111775
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.489568 0.329304 0.03
  rotation 0 1 0 5.30716e-06
  floorSize 4 4
}
DEF Obs1 Solid {
  translation -0.222869 -1.24897 0.142502
  rotation -0.002646061317764138 4.828432404606054e-05 0.9999964980079312 3.12429
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 1 0.05 0.5
      }
    }
  ]
  boundingObject Box {
    size 1 0.05 0.5
  }
}
Robot {
  translation -1.42351 0.713473 0.0696253
  rotation -0.012561928574481563 0.01868416104078558 0.9997465179117612 0.0939271739338515
  children [
    Compass {
    }
    InertialUnit {
      name "imu"
    }
    DEF gps GPS {
      translation -0.05 -0.02 0.03
      rotation 0 0 1 -2.8797953071795863
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.02
            radius 0.02
          }
        }
      ]
      boundingObject Shape {
        appearance PBRAppearance {
        }
        geometry Cylinder {
          height 0.02
          radius 0.02
        }
      }
      physics Physics {
      }
    }
    DEF lidar Lidar {
      translation 0.07 -0.02 0.04
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.025
            radius 0.025
          }
        }
      ]
      boundingObject Shape {
        appearance PBRAppearance {
        }
        geometry Cylinder {
          height 0.025
          radius 0.025
        }
      }
      physics Physics {
      }
      horizontalResolution 32
      fieldOfView 1.5
      verticalFieldOfView 1
      numberOfLayers 1
      maxRange 0.2
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 70990.64725944713
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.05084497543489912 0.06043224674532349 0.000332221520143557
        rotation -0.030538445044510663 0.6998596519028735 -0.7136272633613798 3.1909222798525856
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 58515.2067927689
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.04999849070189097 -0.06000008469764562 1.6288940059832692e-06
        rotation 0.9976051185499567 -0.048908116037727226 0.0489083186029431 1.5731977897684342
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 70990.6500816072
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.049995828790510614 0.060000084994313085 -2.039696395091813e-06
        rotation 0.028602316163787736 -0.7068161756348131 0.706818789627896 3.0844037647689335
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 58515.210959344244
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.049989900198008703 -0.0600038201240265 3.1192958497313516e-06
        rotation 0.9978040654114401 -0.046817038888676256 0.046853088671919 1.5729733425566976
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation 0.1 -0.03 0
      rotation 0 0 1 -0.3
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.1 0.03 0
      rotation 0 0 1 0.3
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  boundingObject USE BODY
  physics Physics {
    mass 1
  }
  controller "controlador"
  window "<none>"
}
Solid {
  translation -1.34884 -0.459107 5.59552e-14
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.9 0.05 0.6
      }
    }
  ]
  name "solid(1)"
  boundingObject Box {
    size 0.9 0.05 0.6
  }
}
Solid {
  translation -1.87071 0.602652 -6.30607e-14
  rotation 2.3464099999870814e-06 -2.3464099999870814e-06 0.9999999999944944 1.5708
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.9 0.05 0.6
      }
    }
  ]
  name "solid(3)"
  boundingObject Box {
    size 0.9 0.05 0.6
  }
}
Solid {
  translation -0.133341 0.401607 -6.30607e-14
  rotation 2.3464099999870814e-06 -2.3464099999870814e-06 0.9999999999944944 1.5708
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.9 0.05 0.6
      }
    }
  ]
  name "solid(4)"
  boundingObject Box {
    size 0.9 0.05 0.6
  }
}
Solid {
  translation -0.941358 1.91592 -1.70087e-13
  rotation 0 0 1 -1.5707953071795862
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.9 0.05 0.6
      }
    }
  ]
  name "solid(6)"
  boundingObject Box {
    size 0.9 0.05 0.6
  }
}
Solid {
  translation -1.70524 2.26021 0.09
  rotation 0 0 -1 4.692820414042842e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(2)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation -2.3761 1.27586 0.09
  rotation 0 0 1 -1.5708053071795867
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(8)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation 1.42442 1.34619 0.09
  rotation 0 0 1 -1.5708053071795867
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(9)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation 1.40695 -0.611603 0.09
  rotation 0 0 1 -1.5708053071795867
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(13)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation -2.39772 -0.707142 0.09
  rotation 0 0 1 -1.5708053071795867
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(7)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation -1.44039 -1.60974 0.09
  rotation 0 0 -1 4.692820414042842e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(10)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation 0.505853 -1.55992 0.09
  rotation 0 0 -1 4.692820414042842e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(11)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation 0.321583 2.25345 0.09
  rotation 0 0 -1 4.692820414042842e-06
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(12)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
Solid {
  translation 0.419486 -0.579864 0.09
  rotation 0 0 1 3.14159
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 2 0.05 0.6
      }
    }
  ]
  name "solid(5)"
  boundingObject Box {
    size 2 0.05 0.6
  }
}
